File: app/src/free/java/com/example/dagger/FlavorModule.java
package com.example.dagger;

import com.example.dagger.di.Free;
import com.example.dagger.di.LiveFeedbackModule;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by LKW on 24/6/2017.
 */

@Module(includes = LiveFeedbackModule.class)
public class FlavorModule {
    private static final String TAG = "FlavorModule";

    @Singleton
    @Provides
    LiveFeedbackManager provideLiveFeedbackManager(@Free LiveFeedbackManager manager) {
        return manager;
    }
}

File: app/src/premium/java/com/example/dagger/FlavorModule.kt
package com.example.dagger

import com.example.dagger.di.LiveFeedbackModule
import com.example.dagger.di.Premium
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by LKW on 25/6/2017.
 */

@Module(includes = arrayOf(LiveFeedbackModule::class))
class FlavorModule {
    @Singleton @Provides
    fun provideLiveFeedbackManager(@Premium manager: LiveFeedbackManager): LiveFeedbackManager {
        return manager
    }
}